// pages/PartnersPage/index.jsx
import React, {useRef} from 'react'
import './PartnersPage.sass'
import FirstScreen from 'component/pageComponents/PartnersPage/FirstScreen'
import Services from 'component/pageComponents/PartnersPage/Services'
import Examples from 'component/pageComponents/PartnersPage/Examples'
import Offer from 'component/pageComponents/PartnersPage/Offer'
import PageFooter from 'component/pageComponents/MainPage/PageFooter'
import {ReactComponent as LogoIcon} from "assets/img/staketabLogo.svg";

export default function PartnersPage(props) {

  const footerConfig = {
    logo: <LogoIcon />,
    copyright: `©${new Date().getFullYear()} - Staketab.com`,
    columns: [
      {
        title: "firstColumn",
        links: [
          {
            title: "home",
            link: "/",
          },
          {
            title: "Staking",
            link: "https://docs.staketab.com/how-to-stake/bitcanna",
          },
          {
            title: "FAQ",
            link: "/#FAQ",
            // ref: faqRef
          },
          {
            title: "Academy",
            link: "https://docs.staketab.com/academy/mina",
          },

        ],
      },
      {
        title: "secondColumn",
        links: [
          {
            title: "Explorer",
            link: "https://mina.staketab.com/",
          },
          {
            title: "Services",
            link: "https://services.staketab.com/",
          },
          {
            title: "About",
            link: "/#company",
            // ref: companyRef
          },
          {
            title: "Partnership",
            link: "/partnership",
          },
        ],
      },
    ],
  };

  return (
    <div className="partnerspage">
      <FirstScreen />
      <Services />
      <Examples />
      <Offer />
      <PageFooter containerClassName="partnerscontainer" classNames={"partnerspageFooter"} inverse links={footerConfig}/>
    </div>
  )
}
