// component/pageComponents/MainPage/SocialLinksBar/index.js
import React from 'react'
import './SocialLinksBar.sass'
import { ReactComponent as TelegramImg } from 'assets/img/Telegram.svg'
import { ReactComponent as DiscordImg } from 'assets/img/Discord.svg'
import { ReactComponent as TwitterImg } from 'assets/img/Twitter.svg'
import { ReactComponent as MailImg } from 'assets/img/Mail.svg'

export default function SocialLinksBar(props) {

  const vertical = {
    flexDirection: 'column'
  }
  
  const horizontal = {
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
  
  const horizontalItems = {
    marginBottom: '0px'
  }
  
  const transparent = {
    backgroundColor: 'transparent',
  }

  const noPadding = {
    padding: 0
  }

  const defineStyle = () => {
    let container = props.horizontal ? horizontal : vertical
    let padding = props.noPadding ? noPadding : {}
    let opacity = props.transparent ? transparent : {}
    let items = props.horizontal ? horizontalItems : {}
    container = {...container, ...opacity, ...padding}
    return {container, items}
  }

  return (
    <aside className="socialLinksBar" style={defineStyle().container}>
      <a href="https://t.me/staketab_ann" className={`socialLinksBar__link ${props.inversed ? 'inversed' : ''}`} style={defineStyle().items}><TelegramImg /></a>
      <a href="https://discord.gg/UBQ85uSyAU" className={`socialLinksBar__link ${props.inversed ? 'inversed' : ''}`} style={defineStyle().items}><DiscordImg /></a>
      <a href="https://twitter.com/staketab" className={`socialLinksBar__link ${props.inversed ? 'inversed' : ''}`} style={defineStyle().items}><TwitterImg /></a>
      <a href="mailto:support@staketab.com" className={`socialLinksBar__link ${props.inversed ? 'inversed' : ''}`} style={defineStyle().items}><MailImg /></a>
    </aside>
  )
}
