// store/MainPage/reducers.js
import {
    SHOW_POPUP,
    SET_POPUP_INFO,
    SET_NET,
    SET_CARDS_INFO
} from './types'

const initialState = {
  isPopupShowed: false,
  popUpInfo: {},
  net: 'mainnet',
  cardsInfo: {
        mainnetCards: [],
        testnetCards: []
  }
}

export function reducerMainPage(state = initialState, action) {
  switch (action.type) {
    case SHOW_POPUP:
        return {
            ...state,
            isPopupShowed: action.payload
        };
    case SET_NET:
        return {
            ...state,
            net: action.payload
        };
    case SET_POPUP_INFO:
        return {
            ...state,
            popUpInfo: action.payload
        };
    case SET_CARDS_INFO:
        return {
            ...state,
            cardsInfo: {
                mainnetCards: action.payload.net === 'mainnet' ? action.payload.data : state.cardsInfo.mainnetCards,
                testnetCards: action.payload.net === 'testnet' ? action.payload.data : state.cardsInfo.testnetCards
            }
        };
    default:
        return state;
  }
}
